### ISSUE TITLE
Request: Add the result file name as an optional parameter

### ISSUE THREAD

**Issue by ak-cube** [cubewise-code/rushti#64]
It would be great if you could send the name of the result file as a parameter in the rushTI execution string. 
This would allow us to create a process for waiting for rushTI completion and reading the result.

The result file could have following columns:
* PID - Process ID of python.exe in the OS (that allows us to connect the result file with rushti.log records)
* Number of Errors
* Stat DateTime
* End DateTime
* Result - Completed/Ended by Timeout/ Canceled

Thanks.

--
Aleksandr

**Comment by MariusWirtz** [comment]
I like this idea. The result file should be machine readable in contrast to the log file which is designed for human consumption.

I would also like to include the runtime in seconds and the number of processes that were run in total.


|PID|Number of Process Runs|Number of Process Fails|Start DateTime      |End DateTime         |Runtime in Seconds|Overall Success|
|---|----------------------|--------------|--------------------|---------------------|------------------|---------------|
|123| 88                   | 0            | 2023-05-31 12:01:01|  2023-05-31 13:01:01| 3600             | True          |

Do you think this format would be friendly enough to consume with TI?

I wonder if we should include a column for processes that succeeded after a retry.

And should we have a separate column for processes that completed with minor errors (= committed) vs processes that aborted (= not committed)

How do you suggest we include the "Ended by timeout error" in the result file? Do you mean the timeout that you can pass to a process execution from TM1py? or some other timeout?

**Comment by ak-cube** [comment]
`Do you think this format would be friendly enough to consume with TI?`
I think that would be enough for tracking history.
If we're talking about troubleshooting or having the full picture of execution, then it would be better to have information about every step in the same format. It could be following records in the same result file or a separate file with detailed information.

`And should we have a separate column for processes that completed with minor errors (= committed) vs processes that aborted (= not committed)`
I think this information is important only on the detail level. In total numbers  it doesn't help much.

`How do you suggest we include the "Ended by timeout error" in the result file? Do you mean the timeout that you can pass to a process execution from TM1py? or some other timeout?`
My idea is to add an additional parameter to the rushTI _config.ini_. 
It should be maximum possible time of rushTI execution in the current environment. That means, if rushTI execution time exceed the limit, then something went wrong and rushTI process should be terminated.
For example, It can help me to prevent the case when I had dead-lock and rushTI was executed 16 hours in the  Production model.

**Comment by ak-cube** [comment]
> I wonder if we should include a column for processes that succeeded after a retry

Sorry, this option in RushTI is still not clear for me. :)
I don't know the cases when I need it. If the process failed once, why it should successfully complete in the next time? Do you have a link (or email) with request for this option?

Thanks.

**Comment by MariusWirtz** [comment]
> > I wonder if we should include a column for processes that succeeded after a retry
> 
> Sorry, this option in RushTI is still not clear for me. :) I don't know the cases when I need it. If the process failed once, why it should successfully complete in the next time? Do you have a link (or email) with request for this option?
> 
> Thanks.

Probably less relevant for TM1 core. Retries are helpful when other systems are fragile and not every request succeeds due to race conditions or bugs in a system. Some of those issues can be resolved through a simple retry.

A realistic scenario would perhaps be when TI tries to read a file that is still blocked by another process, then the TI fails. If the process is retried a second later, it may just succeed.

**Comment by MariusWirtz** [comment]
> My idea is to add an additional parameter to the rushTI _config.ini_.
> It should be maximum possible time of rushTI execution in the current environment. That means, if rushTI execution time exceed the limit, then something went wrong and rushTI process should be terminated.
> For example, It can help me to prevent the case when I had dead-lock and rushTI was exec

Please open a separate issue for this request.

I think it would be easy to do if we implement it as a timeout on individual processes. So you could pass an argument like `max_process_runtime=3600` and if any process hasn't been completed within 3600 seconds, rushti will abort the process as well as the overall tasks execution.

**Comment by MariusWirtz** [comment]
@ak-cube 
Please test this branch: https://github.com/cubewise-code/rushti/pull/68

In the branch you can execute rushti with 5 arguments: 
- "tasks.txt" 
- "2"
- "NORM" 
- "1" 
- "result.csv"

**Comment by MariusWirtz** [comment]
resolved with #68
